geomstats/geometry/matrices_space.py

```diff
@@ -30,6 +30,39 @@ def belongs(self, point):
         _, mat_dim_1, mat_dim_2 = point.shape
         return mat_dim_1 == self.m & mat_dim_2 == self.n
 
+    @staticmethod
+    def mult(a, b):
```

Comment:
NIT: I'm thinking `mul` could be more appropriate, so that users can do mats.mul, which resembles matmul (and not get confused that we have a t and numpy does not)